### PR TITLE
[DC-797] Refactor ajax/DatasetBuilder and ajax/DataRepo 

### PR DESCRIPTION
**Pull request by rjohanek**
### What
- Call DataRepo instead of DatasetBuilder from datasetBuilder, conceptSelector, and DomainCriteriaSelector
- Eliminate DatasetBuilder and its contract
- Update testing to mock DataRepo 

### Why
-  Refactor pages so the cohort builder calls into DataRepo directly. This will reduce overhead when adding a new API.

### To do
- I'm not sure that the tests are as strong as they used to be and I want to better understand jest mocking. 

### Testing strategy
- [ ] Verify that UI still works 
- [ ] Verify tests still cover all functionality

### CODE REVIEW

**Review by snf2ye — APPROVED** [review]

File: src/dataset-builder/ConceptSetCreator.test.ts [hunk 1]
```diff
@@ -7,6 +7,8 @@ import { ConceptSetCreator, toConceptSet } from './ConceptSetCreator';
 import { homepageState } from './dataset-builder-types';
 import { dummyDatasetModel } from './TestConstants';
 
+jest.mock('src/libs/ajax/GoogleStorage');
```
Comment (s-rubenstein) [inline src/dataset-builder/ConceptSetCreator.test.ts:10]: This change seems unrelated, should this have been here before? Or am I missing something about how it connects to the refactor?

File: src/dataset-builder/DatasetBuilder.test.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ import userEvent from '@testing-library/user-event';
 import _ from 'lodash/fp';
 import { h } from 'react-hyperscript-helpers';
 import { DataRepo, DataRepoContract, DatasetModel } from 'src/libs/ajax/DataRepo';
-import { Cohort, ConceptSet, DatasetBuilder, DatasetBuilderContract } from 'src/libs/ajax/DatasetBuilder';
+import { Cohort, ConceptSet } from 'src/libs/ajax/DatasetBuilder';
```
Comment (s-rubenstein) [inline src/dataset-builder/DatasetBuilder.test.ts]: Can we join this line and line 5?

File: src/libs/ajax/DatasetBuilder.ts [hunk 3]
```diff
@@ -1,11 +1,8 @@
 // Types that can be used to create a criteria.
 import _ from 'lodash/fp';
-import { Ajax } from 'src/libs/ajax';
```
Comment (s-rubenstein) [inline src/libs/ajax/DatasetBuilder.ts:3]: Do you think this file should still live in the `ajax` folder? Or should we move it into a libs/utils folder? I think we might want to not have it stored in `src/libs/ajax` anymore if it isn't making ajax calls.

File: src/dataset-builder/ConceptSetCreator.test.ts (hunk 1 above)
Comment (rjohanek) [reply src/dataset-builder/ConceptSetCreator.test.ts:10]: I think it should've been there before. This is required for the initialization of the tests, I'm not sure exactly how they were working before but it might have been fragile because after the refactor, this is needed to run the tests.

File: src/libs/ajax/DatasetBuilder.ts (hunk 3 above)
Comment (rjohanek) [reply src/libs/ajax/DatasetBuilder.ts:3]: yeah good call

File: src/dataset-builder/DatasetBuilder.test.ts (hunk 2 above)
Comment (rjohanek) [reply src/dataset-builder/DatasetBuilder.test.ts]: I'm not sure what you mean

Comment (s-rubenstein) [reply src/dataset-builder/DatasetBuilder.test.ts]: Oh, wait, no, I am wrong! Nevermind, I misread, disregard.

File: src/dataset-builder/CohortEditor.ts [hunk 4]
```diff
@@ -6,21 +6,21 @@ import Slider from 'src/components/common/Slider';
...
   Cohort,
   CriteriaGroup,
-  DatasetBuilder,
   LoadingAnyCriteria,
   ProgramDataListCriteria,
   ProgramDataListValue,
   ProgramDataRangeCriteria,
-} from 'src/libs/ajax/DatasetBuilder';
-import colors from 'src/libs/colors';
+} from 'src/libs/util/DatasetBuilder';
```
Comment (s-rubenstein) [inline src/dataset-builder/CohortEditor.ts]: Optional nit: I would probably say that a better path might be `src/dataset-builder/DatasetBuilderUtils`? That might align more with the codebase's goal of isolating feature areas.

**Review by s-rubenstein — APPROVED** [review]
LGTM, pending the one comment.
